extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java

```diff
@@ -54,6 +54,12 @@ public class ProbeConfig {
     @ConfigItem
     Optional<String> grpcAction;
 
+    /**
+     * If enabled and `grpc-action` is not provided, it will use the generated service name and the gRPC port.
+     */
+    @ConfigItem(defaultValue = "false")
+    boolean grpcActionEnabled;
```

Comment:
I am wondering  if there are ways to infer this configuration value using a  build item or something like that.
I would like to avoid having `grpcAction` and `grpcActionEnabled` as it can be confusing.

If this is not possible then we need to use a name that clearly suggests that by setting this to `true` the grpcAction will be infered. But I would prefer if we could somehow completly avoid having a new property.

Reply:
For now we can leave it as is. I created: https://github.com/quarkusio/quarkus/issues/33583 as a general solution / approach for such cases.